src/Groups/cosets.jl

```diff
@@ -50,8 +56,11 @@ function right_coset(H::GAPGroup, g::GAPGroupElem)
 end
 
 """
-    right_coset(H::Group, g::GAPGroupElem)
+    left_coset(H::Group, g::GAPGroupElem)
+    *(g::GAPGroupElem, H::Group)
 Return the coset `gH`.
+!!! note
+    Since GAP supports right cosets only, the underlying GAP object of `left_coset(H,g)` is the right coset `H^(g^-1) * g`.
```

Comment:
While this describes what the code does, this strikes me as deeply problematic. It means that you have to creates lots of new subgroups by conjugating H around.

My idea would be to represent `gH` by `Hg^{-1}`, using that fact that the inversion map induces a bijection between them.

That said, I am actually still sceptical about the whole idea of providing left cosets. Perhaps it's a good idea to do so, but I'd really like to see some concrete applications in order to figure out what the "right" way (or at least a "somewhat useful way") to go about them might be.

Reply:
> My idea would be to represent gH by Hg^{-1}, using that fact that the inversion map induces a bijection between them.

This would be much faster during the creation of the coset (I don't have to create the conjugate subgroup), but would it be slower every time I need to access to it?
If `lc=gH`, and I save it as `Hg^-1`, every time I type something like `rand(lc)` or `elements(lc)`, or some intersection involving `lc`, I need to compute an inverse. A possible syntax for the `rand` function could be
`> function rand(lc::Coset)`
`> x=rand(lc)`
`> if isleft(lc) then x=x^-1 end`
`> return x`
`> end`

Reply:
I think it is perfectly realizable, maybe with a little care about the fact that the underlying GAP object is not the object I'm interested (but its image under the inverse function), but is it really faster?

Reply:
Well, that exactly is my point: what is "faster" depends entirely on the use case. But no matter how you implement the left cosets, one thing stays the same: as long as they simply (ab)use GAP's right cosets, it will always be faster if you rewrite your code to use right cosets instead of left cosets.

Anyway: as long as this is an implementation detail is hidden from the user, this is fine. But I would not want to mention it to the user! The user should not have to care about this.

Next thing: GAP doesn't really do much with `RightCosets` anyway. So we could think about simply not using GAP objects of type `RightCoset` at all, and simply implement our own two types `RightCoset` and `LeftCoset`, which simply wrap a group element `g` plus a subgroup `H`. Implementing things like `rand` or `elements` then is easy: e.g. `rand(H) * g` to get a random element. I'll put this idea into another issue

Reply:
It seems good. Do I try this new type on the same branch? Or do we open a new one?